File: processor/src/main/java/io/deloop/tools/reference/replay/helpers/MethodIdGenerator.java
package io.deloop.tools.reference.replay.helpers;

import static java.util.UUID.randomUUID;

public interface MethodIdGenerator {

    String uniqueId(String methodName);

    MethodIdGenerator REAL = new MethodIdGenerator() {
        @Override
        public String uniqueId(String methodName) {
            return methodName + "-" + randomUUID().toString();
        }
    };
}

File: processor/src/main/java/io/deloop/tools/reference/replay/helpers/MethodIdGenerator.kt
package io.deloop.tools.reference.replay.helpers

import java.util.UUID.randomUUID

interface MethodIdGenerator {

    fun uniqueId(methodName: String): String

    companion object {
        val real: MethodIdGenerator = object : MethodIdGenerator {
            override fun uniqueId(methodName: String): String {
                return methodName + "-" + randomUUID().toString()
            }
        }
    }
}
